packages/flutter/test/material/tabs_test.dart

```diff
@@ -2350,4 +2350,76 @@ void main() {
     expect(find.text('Tab1'), findsOneWidget);
     expect(find.text('Tab2'), findsOneWidget);
   });
+
+  testWidgets('DefaultTabController should allow for a length of zero', (WidgetTester tester) async {
+    List<String> tabTextContent = <String>[];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return DefaultTabController(
+              length: tabTextContent.length,
+              child: Scaffold(
+                appBar: AppBar(
+                  title: const Text('Default TabBar Preview'),
+                  bottom: tabTextContent.isNotEmpty
+                    ? TabBar(
+                      isScrollable: true,
+                      tabs: tabTextContent.map((String textContent) {
```

Comment:
It totally doesn't matter, but. This could be a one-liner:
```dart
tabs: tabTextContent.map((String text) =>  Tab(text: text)).toList(),
```
In the same vein:
```dart
bottom: tabTextContent.isEmpty ? null : TabBar(
  isScrollable: true,
  tabs: tabTextContent.map((String text) =>  Tab(text: text)).toList(),
),
```